spark/v3.1/spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java

```diff
@@ -1469,12 +1484,18 @@ public void testPartitionsTableLastUpdatedSnapshot() {
         new GenericRecordBuilder(
             AvroSchemaUtil.convert(
                 partitionsTable.schema().findType("partition").asStructType(), "partition"));
+
+    List<DataFile> dataFiles = dataFiles(table);
+    Assert.assertEquals("Table should have 3 data files", 3, dataFiles.size());
+    assertDataFilePartitions(dataFiles, new int[] {1, 2, 2});
```

Comment:
Sorry I'm trying to remember why I said the previous comment, I was probably thinking varargs, like 
`assertDataFilePartitions(List<DataFile>, int... partitions)` so then it would just be ` assertDataFilePartitions(dataFiles, 1,2,3)`, but on second look it may not be so clean.
 
I think Arrays.asList() is equally long, let's revert to that, and it may make the length check easier inside the method?